src/methods/clipping/clipping_processor.jl

```diff
@@ -0,0 +1,281 @@
+            step = f_step(curr.ent_exit, on_a_list)
+            curr_not_intr = true
+            while curr_not_intr
+                # Traverse polygon either forwards or backwards
+                idx += step
+                idx = (idx > curr_npoints) ? mod(idx, curr_npoints) : idx
+                idx = (idx == 0) ? curr_npoints : idx
+
+                # Get current node and add to pt_list
+                curr = curr_list[idx]
+                push!(pt_list, curr.point)
+                if curr.inter 
+                    # Keep track of processed intersection points
+                    curr_not_start = curr != start_pt && curr != b_list[start_pt.neighbor]
+                    if curr_not_start
+                        processed_pts += 1
+                        a_idx_list[curr.idx] = 0
+                    end
+                    curr_not_intr = false
+                end
+            end
+
+            # Switch to next list and next point
+            curr_list, curr_npoints = on_a_list ? (b_list, n_b_pts) : (a_list, n_a_pts)
+            on_a_list = !on_a_list
+            idx = curr.neighbor
+            curr = curr_list[idx]
+        end
+        push!(return_polys, GI.Polygon([pt_list]))
+    end
+    return return_polys::Vector{GI.Polygon}
+end
+
+#=
+    _add_holes_to_polys!(::Type{T}, return_polys, hole_iterator)
+
+The holes specified by the hole iterator are added to the polygons in the return_polys list.
+If this creates more polygon, they are added to the end of the list. If this removes
+polygons, they are removed from the list
+=#
+function _add_holes_to_polys!(::Type{T}, return_polys, hole_iterator) where T
+    n_polys = length(return_polys)
+    # Remove set of holes from all polygons
+    for i in 1:n_polys
+        n_new_per_poly = 0
+        for hole in hole_iterator  # loop through all holes
+            hole_poly = GI.Polygon([hole])
+            # loop through all pieces of original polygon (new pieces added to end of list)
+            for j in Iterators.flatten((i:i, (n_polys + 1):(n_polys + n_new_per_poly)))
+                if !isnothing(return_polys[j])
+                    new_polys = difference(return_polys[j], hole_poly, T; target = GI.PolygonTrait)
+                    n_new_polys = length(new_polys)
+                    if n_new_polys == 0  # hole covered whole polygon
+                        return_polys[j] = nothing
+                    else
+                        return_polys[j] = new_polys[1]  # replace original
+                        if n_new_polys > 1  # add any extra pieces
+                            append!(return_polys, @view new_polys[2:end])
+                            n_new_per_poly += n_new_polys - 1
+                        end
+                    end
+                end
+            end
+        end
+        n_polys += n_new_per_poly
+    end
+    # Remove all polygon that were marked for removal
+    filter!(!isnothing, return_polys)::Vector{GI.Polygon}
```

Comment:
This type assert is for an abstract type

Reply:
Oops. I guess I was trying to make sure it knew it wasn't a type union with `Nothing`. If we filter out the `nothing` values, is that something we need to worry about?

Reply:
Maybe try try spell out the full type?

But julia might work it out anyway. Profiling/Cthulhu should show it

Reply:
I didn't see any slowdowns